src/RevisionPopupWidget.js

```diff
@@ -48,29 +51,52 @@ RevisionPopupWidget.prototype.show = function ( data, $target ) {
 		// We typically link to Special:Contribs for IPs.
 		userPageUrl = isIP ? contribsUrl : mw.util.getUrl( `User:${data.username}` ),
-		userLinks = `
-			<a href="${userPageUrl}">${username}</a>
-			${mw.msg( 'parentheses-start' )}<a href="${mw.util.getUrl( `User talk:${username}` )}">${mw.msg( 'talkpagelinktext' )}</a>
-			${mw.msg( 'pipe-separator' )}
-			<a href="${contribsUrl}">${mw.msg( 'contribslink' )}</a>${mw.msg( 'parentheses-end' )}
-		`,
+		// Create links using the jQuery objects so they're properly escaped
+		$userLinks = $( [] )
+			.add(
+				$( '<a>' )
+					.attr( 'href', userPageUrl )
+					.append( Tools.bidiIsolate( username ) )
+			)
+			.add( document.createTextNode( ' ' + mw.msg( 'parentheses-start' ) ) )
+			.add(
+				// Talk page
+				$( '<a>' )
+					.attr( 'href', mw.util.getUrl( `User talk:${username}` ) )
+					.text( mw.msg( 'talkpagelinktext' ) )
+			)
+			.add( document.createTextNode( ' ' + mw.msg( 'pipe-separator' ) + ' ' ) )
+			.add(
+				$( '<a>' )
+					.attr( 'href', contribsUrl )
+					.text( mw.msg( 'contribslink' ) )
+			)
+			.add( document.createTextNode( mw.msg( 'parentheses-end' ) ) ),
 		dateStr = moment( data.revisionTime ).locale( mw.config.get( 'wgUserLanguage' ) ).format( 'LLL' ),
-		diffLink = `<a href="${mw.util.getUrl( `Special:Diff/${data.revisionId}` )}">${dateStr}</a>`,
-		addedMsg = mw.message( 'ext-whowrotethat-revision-added', userLinks, diffLink ).parse(),
-		commentMsg = data.comment ? `<span class="comment comment--without-parentheses ext-wwt-revisionPopupWidget-comment">${data.comment}</span>` : '',
+		// Use jQuery to make sure attributes are properly escaped
+		$diffLink = $( '<a>' )
+			.attr( 'href', mw.util.getUrl( `Special:Diff/${data.revisionId}` ) )
+			.text( dateStr ),
+		addedMsg = mw.message( 'ext-whowrotethat-revision-added', $userLinks, $diffLink ).parse(),
+		commentMsg = data.comment ?
+			`<span class="comment comment--without-parentheses ext-wwt-revisionPopupWidget-comment">${Tools.bidiIsolate( $.parseHTML( data.comment ), true )}</span>` :
+			'',
 		sizeMsg = data.size ? getSizeHtml( data.size ) : '',
 		attributionMsg = `<div class="ext-wwt-revisionPopupWidget-attribution">${mw.message( 'ext-whowrotethat-revision-attribution', data.score ).parse()}</div>`,
-		$popupContent = $( '.ext-wwt-revisionPopupWidget-content' ),
 		html = $.parseHTML( `${addedMsg.trim()} ${commentMsg}${sizeMsg} ${attributionMsg}` );
 
-	$popupContent.html( html );
+	this.$popupContent.html( html );
 
 	if ( $target.find( '.thumb' ).length ) {
 		$target = $target.find( '.thumb' );
 	}
 
 	// Make sure all links in the popup (including in the edit summary) open in new tabs.
-	$popupContent.find( 'a' ).attr( 'target', '_blank' );
+	this.$popupContent.find( 'a' )
+		.attr( 'target', '_blank' )
+		// Fix security volunerability associated with target=_blank
+		// see example: https://developers.google.com/web/tools/lighthouse/audits/noopener
+		.attr( 'rel', 'noopener noreferrer' );
```

Comment:
Interesting! If this is a real concern, our other tools (Event Metrics, XTools, Pageviews, etc.) are littered with the same vulnerability. For WWT these should be internal links though, right? Anchor tags are not generated for URLs in edit summaries. It seems like good practice nonetheless, but I would at least remove `noreferrer` since we probably have no reservations about this for internal links, and it helps analytics track where traffic came from.

I'm probably missing something, but I don't believe that `noreferrer` works either, because I almost always see a wiki URL as the referrer in the access logs for XTools/Pageviews.

Reply:
Yup we might want to consider doing that in other places. This was all over twitter a couple of weeks ago, I don't think a lot of people know about it -- but it's pretty huge.

The recommendation is to add noreferrer but I think Firefox doesn't recognize it, so noopener is the alternative to get better support.

Here's an interesting article about the potential "attack" vector: https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/

I wouldn't say it's an urgent horrible thing to fix, but it might be a good reinforcement to add whenever possible.

Reply:
Firefox 52 according to MDN. But at any rate, I'm more interested if these should apply here, since they are internal links? I noticed the wiki does not use this for internal links.

Side note, all external links on the wiki have `noopener` and `noreferrer`, but I definitely still see referrer data in my tools' access logs saying they came from the wiki, and even for the latest browser versions. Maybe they're spoofing, who knows...

Reply:
I'll certainly get this added to XTools, which has many an external link. Pageviews and Event Metrics are probably fine.

Reply:
Hm that's a good point, if they're strictly internal links, always, then you're right, it's probably an overkill, I didn't think about the fact that we know 100% that the links are always internal.

Reply:
Did we want to remove it, then? I don't think it's overkill, just curious if we wanted to match MediaWiki.

Reply:
Yeah, I am not sure in this case. We're talking about internal links we control, so it's probably okay. If this wasn't a gadget or extension then the problem of google not indexing it would've actually been a problem, but that's not really the case here anyways.

So, yeah, I think this is a case where it's not actually crucial either way -- and we can probably remove it. I'll do that in the next patch fix.

BTW, To be fair, just specifically the "match" to MediaWiki, I think we (generally) might want to do it the other way around and start adding those to relevant (non internal) places in MW if/when we do target=_blank (which is already not common). Anyways, that's totally an aside. I'll raise it to Volker.